e2e-tests/spring-rest-openapi-v3/src/main/kotlin/com/foo/rest/examples/spring/openapi/v3/authenticatedswaggerendpoint/AuthenticatedSwaggerEndpointRest.kt

```diff
@@ -0,0 +1,54 @@
+package com.foo.rest.examples.spring.openapi.v3.authenticatedswaggerendpoint
+
+import org.springframework.http.MediaType
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+import java.io.File
+
+
+@RestController
+class AuthenticatedSwaggerEndpointRest {
+
+    private val SECRET = "Secret Authentication Bearer..."
+
+    // OD - Note, this path will work for e2e test but not for manual test
+    // For manual testing, the working directory should be set to ./e2e-tests/spring-rest-openapi-v3/
+    private val SWAGGER_PATH = "./src/main/resources/static/authenticatedSwaggerEndpointOpenAPIv3.json"
+
+    @PostMapping(path = ["/api/logintoken/login"], consumes = [MediaType.APPLICATION_JSON_VALUE])
+    fun login(@RequestBody login : LoginDto) : ResponseEntity<AuthDto>{
+
+        if(login.userId == "userName1" && login.password == "password1234"){
+            return ResponseEntity.ok(AuthDto("foo", TokenDto(SECRET)))
+        }
+
+        return ResponseEntity.status(400).build()
+    }
+
+    @GetMapping(path = ["/api/logintoken/check"])
+    fun check(@RequestHeader("Authorization") authorization: String?) : ResponseEntity<String>{
+
+        if(authorization == "Bearer $SECRET"){
+            return ResponseEntity.ok("OK")
+        }
+
+        return ResponseEntity.status(401).build()
+    }
+
+    @GetMapping(path = ["/v3/api-docs"])
```

Comment:
are we sure this is called, and not conflicting with endpoint added by SpringDoc?
maybe a better option would be to use a different endpoint path, and then modify `getProblemInfo()` accordingly

Reply:
Hi Andrea,

I have tried this endpoint by testing it manually using Postman and I have also tried if EM tests can generate test cases for the endpoint, which requires them to access the swagger in the black-box. Both cases worked, so I assume calling /v3/api-docs works. I indeed had to override the endpoint since I wanted to create an authenticated swagger for testing purposes.

Reply:
still, overriding an endpoint is risky... eg a new version of Spring could change the algorithm that decides which one is used. 
rename this endpoint with something different, eg `/openapi-schema`. and then update controller